client/src/pages/add-recipe/add-recipe-manual.page.js

```diff
@@ -15,14 +20,71 @@ const ManualInsertPage = () => {
     const navigate = useNavigate();
     const scrollOffset = -1 * window.innerHeight * 0.1;
 
+    // CHATGPT --->
+    const [ingredients, setIngredients] = useState([{ value: "" }]);
+    const [steps, setSteps] = useState([{ value: "" }]);
+
+    const handleIngChange = (index, event) => {
+      const values = [...ingredients];
+      values[index].value = event.target.value;
+      setIngredients(values);
+    };
+    const handleStepChange = (index, event) => {
+        const values = [...steps];
+        values[index].value = event.target.value;
+        setSteps(values);
+      };
+  
+    const handleAddIng = () => {
+      const values = [...ingredients];
+      values.push({ value: "" });
+      setIngredients(values);
+    };
+    const handleAddStep = () => {
+        const values = [...steps];
+        values.push({ value: "" });
+        setSteps(values);
+      };
+  
+    const handleRemoveIng = (index) => {
+      const values = [...ingredients];
+      values.splice(index, 1);
+      setIngredients(values);
+    };
+    const handleRemoveStep = (index) => {
+      const values = [...steps];
+      values.splice(index, 1);
+      setSteps(values);
+    };
+
+    // <--- CHATGPT 
+
     const handleSubmit = (event) => {
         event.preventDefault();
         const data = new FormData(event.currentTarget);
+        
+        console.log(ingredients);
+        console.log(steps);
+        
+        // need to handle ingredients, steps
+        let ing_string = ""
+        for (let ing of ingredients) {
+            if (ing.value !== '' && ing.value !== '\n') {
+                ing_string = ing_string + ing.value + "\n";
+            }
+        }
+
+        let step_string = ""
```

Comment:
Similar note to above for the steps. Look at the code I've written and try and get it to match the backend functionality.